chainio/clients/builder.go

```diff
@@ -36,15 +37,19 @@ type BuildAllConfig struct {
 // for non-instrumented clients that doesn't return metrics/reg, and another instrumented-constructor
 // that returns instrumented clients and the metrics/reg.
 type Clients struct {
-	AvsRegistryChainReader     *avsregistry.AvsRegistryChainReader
-	AvsRegistryChainSubscriber *avsregistry.AvsRegistryChainSubscriber
-	AvsRegistryChainWriter     *avsregistry.AvsRegistryChainWriter
-	ElChainReader              *elcontracts.ELChainReader
-	ElChainWriter              *elcontracts.ELChainWriter
-	EthHttpClient              eth.Client
-	EthWsClient                eth.Client
-	Metrics                    *metrics.EigenMetrics // exposes main avs node spec metrics that need to be incremented by avs code and used to start the metrics server
-	PrometheusRegistry         *prometheus.Registry  // Used if avs teams need to register avs-specific metrics
+	AvsRegistryChainReader      *avsregistry.AvsRegistryChainReader
+	AvsRegistryChainSubscriber  *avsregistry.AvsRegistryChainSubscriber
+	AvsRegistryChainWriter      *avsregistry.AvsRegistryChainWriter
+	ElChainReader               *elcontracts.ELChainReader
+	ElChainWriter               *elcontracts.ELChainWriter
+	EthHttpClient               eth.Client
+	EthWsClient                 eth.Client
+	Wallet                      wallet.Wallet
+	TxManager                   txmgr.TxManager
+	AvsRegistryContractBindings *utils.AvsRegistryContractBindings
+	EigenlayerContractBindings  *utils.EigenlayerContractBindings
```

Comment:
@shrimalmadhur any oppositions to returning everything but the kitchen sink in this constructor?
I was thinking we break it up into smaller builders (for eg one for high-level chainio reader/writer/subscribers, and another for low-level interactions), but I don't see any problem with returning everything in here and letting user take whatever they need?

Reply:
let me know if you prefer break it, i will adjust my PR, thanks!

Reply:
IMO, 
we should have 3 exported Methods
`BuildELClients` - just returns core EL contract Bindings
`BuildAVSRegistryClients` - just returns AVS resgistry bindings
`BuildAllClients` - returns all
@samlaf how about this and then user can use based on whatever is needed? I don't want someone to force get AVS registry if not needed.

Reply:
https://github.com/Layr-Labs/eigensdk-go/pull/117 this PR does this. I think we can merge that one first then and rebase this one on top?

Reply:
yes +1 to that.

Reply:
Oh really sorry, i didn't notice that there is a PR already! Just make my PR more clean...

Reply:
I am totally okay you can go with that one first!

Reply:
@renlulu no worries, that was an old one... just merged the other PR. Can you rebase here?

Reply:
GM, just rebased!